assets/src/blocks/Gallery/GalleryGrid.js

```diff
@@ -1,16 +1,37 @@
-export const GalleryGrid = ({ images }) => (
-  <div className="container">
+import { useEffect, useState } from '@wordpress/element';
+import { Lightbox, wpImagesToItems } from '../../components/Lightbox/Lightbox';
+
+export const GalleryGrid = ({ images }) => {
+  const [lightboxIsOpen, setLightboxIsOpen] = useState(false);
+  const [lightboxIndex, setLightboxIndex] = useState(0);
+
+  const onImageClick = lightboxIndex => {
+    setLightboxIsOpen(true);
+    setLightboxIndex(lightboxIndex);
+  };
+
+  const onLightboxClose = () => {
+    setLightboxIsOpen(false);
+  };
+
+  const items = wpImagesToItems(images);  
+
+  return <div className="container">
     <div className="grid-row">
-      {images.map(image => (
+      {images.map((image, index) => (
         <div key={image.image_src} className="grid-item">
           <img
             src={image.image_src}
             srcSet={image.image_srcset}
             style={{ objectPosition: image.focus_image }}
             alt={image.alt_text}
+            onClick={() => {
+              onImageClick(index);
+            }}
           />
         </div>
       ))}
     </div>
+    <Lightbox isOpen={lightboxIsOpen} index={lightboxIndex} items={items} onClose={onLightboxClose} />
```

Comment:
When the lightbox calls `onClose`, the only thing that happens is the state variable `lightboxIsOpen` is set to false. But `lightboxIsOpen` is then only used to pass back into the lightbox.

Would it be possible to handle that internally in `Lightbox`?

Reply:
So I went around this and we actually need that callback in the parent to toggle `isOpen` in the parent, so it reopens the Lightbox even if  `items` or `index` stays the same (e.g.: you close it and click the same image).

Reply:
There's maybe a way around that, but maybe it's not worth digging in further, as there's not too many blocks using a lightbox. So I'm ok with the current version.

But it feels like there must be a way to create a Lightbox component that can be used without duplicating [this entire block](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/076c55d83ecb2a8c50bb09c160d4b123937cc42c/assets/src/blocks/Gallery/GalleryFrontend.js#L26-L36) in each place.

Reply:
Maybe a `useLightbox` hook could encapsulate that block of code?

Reply:
Then `setLightboxIsOpen` and `setLightboxIndex` would be just internal of that hook, it only would need to expose the other things.

Reply:
The Lightbox itself doesn't need that block: 
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/414/files#diff-5807ab99eebdcdb5559e04f7ee7d3fcff716a376e2b33ca0664fefc9fc19dfc0R28

That's just what GalleryFrontend does, not a hard requirement to use the Lightbox, it just needs the props. How you manage those props outside of it it's up to you, it can be different depending on the case.

Reply:
Wouldn't it need all the same things if another block also creates a gallery? Besides I don't think it hurts to add the hook even if there's only a single usage, as it greatly improves readability imo.

Reply:
The custom hook sounds good. As for your question, I think it does not need _exactly_ that, it could be handled differently. The Lightbox doesn't care.

Reply:
Indeed the lightbox doesn't require it, but I rather mean if another block would use it the same way as Gallery, it would probably be needing the same things.

Reply:
Right, so I extracted the block as a custom hook in a separate file, so it's reusable in other places.

Reply:
:100: